### ISSUE TITLE
Using return value of a function that returns None is not always reported.

### ISSUE THREAD

**Issue by NielsPT** [pylint-dev/pylint#10476]
### Bug description

```python
def f1(i: int) -> None:
    print(i)


def f2() -> None:
    r = f1(2)  # assignment-from-no-return
    if not r:
        print("error")
    if not f1(2):  # no error
        print("error")
```

### Configuration

```ini

```

### Command used

```shell
pylint experiments/type_test.py -E
```

### Pylint output

```python
experiments/type_test.py:6:4: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
```

### Expected behavior

In the code snippet, line 6 will report an error, but line 9 doesn't. I'd expect to see an error on line 9, since the code essentially does the same thing.
Not sure whether this is by design or a bug. If it's by design, feel free to change this issue to a feature request. 

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.10
Python 3.13.5 (main, Jun 11 2025, 15:36:57) [Clang 17.0.0 (clang-1700.0.13.3)]
```

### OS / Environment

MacOS 15.5

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
We check this in ``visit_assign`` here:

https://github.com/pylint-dev/pylint/blob/fccbf982b95483bd166ada178c4ac9883cfa6907/pylint/checkers/typecheck.py#L1216

We could also do this check in a boolean context where the call is used immediately (maybe using another message like ``condition-evals-to-constant`` ?).

**Comment by Vimal-Shady** [comment]
Hi! 👋 I’d love to work on this issue. Could you please assign this to me? Thank you! 😊

**Comment by Pierre-Sassoulas** [comment]
Sure :) what are your thoughts on the specifications ?

**Comment by lfjxj03** [comment]
I checked this code with mypy, and it does report an error on line 9. The error message is as follows:
error: "f1" does not return a value (it only ever returns None)  [func-returns-value]

**Comment by kaidenwang0309-hub** [comment]
I think we should have the implemented warning message be `assignment-from-no-return` because it describes the root issue of the syntax, rather than `condition-evals-to-constant`, which is more general.

**Comment by SumanGouda** [comment]
Hi everyone! 
I saw this issue was open and inactive, so I went ahead and implemented a fix with full test coverage.

I have submitted a Pull Request to resolve this here: #11168. Ready for review!"
